test/e2e/framework/providers/openstack/openstack.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstack
+
+import (
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+func init() {
+	framework.RegisterProvider("openstack", newProvider)
```

Comment:
this provider does nothing, this is the same as `skeleton`, why would we not just use skeleton?

Reply:
@BenTheElder That is a nice point.
Some storage e2e tests contain cloud provider specific tests like https://github.com/kubernetes/kubernetes/blob/f7a6b0a8602e02d67fabaa85458a94c0f14599a5/test/e2e/storage/volume_provisioning.go#L375
If using `skeleton`, we cannot operate these OpenStack specific tests.
That is a reason why I proposed this PR.

Reply:
this hits the long-term planning problem of how to break out providers from the e2e testing framework too. for now i'd say add the nullprovider = openstack back.

Reply:
Yeah, that also is a nice point. I guess we might need to implement cloud-provider specific e2e tests on each cloud-provider repo in long-term.